export/mlab_export_test.py

```diff
@@ -0,0 +1,440 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for collectd-mlab."""
+
+import logging
+import os
+import StringIO
+import unittest
+
+# Third-party modules.
+import mock
+
+# Module under test.
+import mlab_export
+
+# Too many public methods. Hard to avoid for unit tests.
+# pylint: disable=R0904
+
+# Static timestamp for unittests.
+FAKE_TIMESTAMP = 1410001000
+logging.basicConfig(filename='/dev/null')
+
+
+def disable_show_options(options):
+  """Sets all show options to False."""
+  options.show_nagios = False
+  options.show_rrdfile = False
+  options.show_rawmetric = False
+  options.show_metric = False
+  options.show_skipped = False
+  return options
+
+
+def enable_show_options(options):
+  """Sets all show options to True."""
+  options.show_nagios = True
+  options.show_rrdfile = True
+  options.show_rawmetric = True
+  options.show_metric = True
+  options.show_skipped = True
+  return options
+
+
+class MlabExport_GlobalTests(unittest.TestCase):
+
+  def setUp(self):
+    self._testdata_dir = os.path.join(
+        os.path.dirname(mlab_export.__file__), 'testdata')
+
+  def testcover_init_global(self):
+    mlab_export.init_global()
+
+  def testunit_LockFile_acquire_WHEN_locked_RAISES_IOError(self):
+    fake_tstamp = os.path.join(self._testdata_dir, 'fake.tstamp')
+
+    lock1 = mlab_export.LockFile(fake_tstamp)
+    lock1.acquire()
+    lock2 = mlab_export.LockFile(fake_tstamp)
+
+    self.assertRaises(IOError, lock2.acquire)
+    lock1.release()
+
+  def testunit_get_mtime_RETURNS_mtime(self):
+    fake_tstamp = os.path.join(self._testdata_dir, 'fake.tstamp')
+    os.utime(fake_tstamp, (FAKE_TIMESTAMP, FAKE_TIMESTAMP))
+    
+    returned_tstamp = mlab_export.get_mtime(fake_tstamp)
+
+    self.assertEqual(FAKE_TIMESTAMP, returned_tstamp)
+
+  def testunit_get_mtime_WHEN_timestamp_file_created_RETURNS_zero(self):
+    fake_tstamp = os.path.join(self._testdata_dir, 'no_such_file.tstamp')
+    # Make sure that fake timestamp file actually doesn't exist.
+    if os.path.exists(fake_tstamp):
+      os.remove(fake_tstamp)
```

Comment:
Yes, that's a good idea. Since this clean up does not apply to any of the other tests here, I moved this one to a separate test class.  9641b9aced8d8be7d8362fce14c598b52a02c17a